opal/mca/common/ucx/common_ucx.c

```diff
@@ -203,21 +204,23 @@ static bool opal_common_ucx_check_device(const char *device_name, char **device_
 {
     char sysfs_driver_link[PATH_MAX];
     char driver_path[PATH_MAX];
-    char *ib_device_name;
+    char ib_device_name[NAME_MAX];
     char *driver_name;
     char **list_item;
     ssize_t ret;
+    char *ib_device_name_fmt;
 
+    opal_asprintf(&ib_device_name_fmt, "%%%u[^:]%%*d", NAME_MAX-1);
     /* mlx5_0:1 */
-    ret = sscanf(device_name, "%m[^:]%*d", &ib_device_name);
+    ret = sscanf(device_name, ib_device_name_fmt, &ib_device_name);
+    free(ib_device_name_fmt);
```

Comment:
maybe ib_device_name_fmt could be fixed size on the stack, so no need to release